### PR TITLE
Loosen up click constraint

### PR DESCRIPTION
**Pull request by eapolinario**
Signed-off-by: Eduardo Apolinario <eapolinario@users.noreply.github.com>

# TL;DR
`click` stable version reached 8.0 recently. Time to bump that requirement to 9.0

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
The upper bound constraint in the version of `click` has been set to `8.0` for a [long time](https://github.com/tnsetting/flytekit/commit/60737e306d72f05d07de2d4598b8e6a281176c0e#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7), but `click` has recently launched version `8.0`. We're starting to see some of our dependencies require at least 8.0 (for example, `papermill` requires `black` which just recently set the lower bound to ['>=8.0.0'](https://github.com/psf/black/blame/main/setup.py#L100)).
 

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
         "wheel>=0.30.0,<1.0.0",
         "pandas>=1.0.0,<2.0.0",
         "pyarrow>=4.0.0,<7.0.0",
-        "click>=6.6,<8.0",
+        "click>=6.6,<9.0",
```
Comment (wild-endeavor) [inline setup.py:42]: can we just remove the upper bound?

Comment (eapolinario) [reply setup.py:42]: we could. I guess the idea for upper-bounds is to catch the case where the API changes between major versions, but I'm not sure if we need this kind of protection for `click`.

**Review by wild-endeavor — APPROVED** [review]

**Comment by eapolinario** [comment]
Superseded by https://github.com/flyteorg/flytekit/pull/838.
